product_country_restriction/models/country_restriction_mixin.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2018 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class CountryRestrictionMixin(models.AbstractModel):
+
+    _name = "country.restriction.mixin"
+    _description = "Country Restriction Mixin"
+
+    def _has_country_restriction(self, countries, date=False):
+        """
+        For one product, get if it has country embargo
+        :param countries:
+        :param date:
+        :return: boolean
+        """
+        self.ensure_one()
+        if not date:
+            date = fields.Date.today()
+        res = self._get_country_restrictions(countries, date=date)
+        if self in res:
+            return res
+        return {}
+
+    def _get_country_restrictions(self, countries, date=False, restriction_id=False):
+        if not date:
+            date = fields.Date.today()
+        product_by_country = {}
+        for country in countries:
+            product_by_country.update({country: self})
+        res = self.env["product.country.restriction"]._get_restriction(
+            product_by_country=product_by_country,
+            date=date,
+            restriction_id=restriction_id,
+        )
+        res = self._update_country_strategy(res)
+        return res
+
+    def _update_country_strategy(self, result):
+        """
+        This is used to apply another strategy to the rules result
+        depending on company strategy
+        :return:
+        """
+        strategy = self.env.user.company_id.country_restriction_strategy
```

Comment:
IMO it's not correct: it should be `self.env.company.country_restriction_strategy`. And maybe check before if the current record has a `company_id` (and use it in priority).
Something like
```
company = self.env.company
if hasattr(self, 'company_id') and self.company_id:
    company = self.company_id
strategy = company.country_restriction_strategy
```